manager.go

```diff
@@ -0,0 +1,220 @@
+package sandbox
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"net/http"
+
+	"github.com/bmizerany/pat"
+	"github.com/dchest/uniuri"
+	"gopkg.in/vinxi/layer.v0"
+	"gopkg.in/vinxi/vinxi.v0"
+)
+
+type Layer struct {
+	layer *layer.Layer
+}
+
+type Options struct {
+	Optional bool
+}
+
+type Rule interface {
+	ID() string
+	Name() string
+	Description() string
+	// Options() Options
+	JSONConfig() string
+	Match(*http.Request) bool
+}
+
+type Scope struct {
+	disabled bool
+	rules    []Rule
+	plugins  *PluginLayer
+
+	ID          string
+	Name        string
+	Description string
+}
+
+func NewScope(rules ...Rule) *Scope {
+	return &Scope{ID: uniuri.New(), Name: "default", plugins: NewPluginLayer(), rules: rules}
+}
+
+func (s *Scope) UsePlugin(plugin Plugin) {
+	s.plugins.Use(plugin)
+}
+
+func (s *Scope) AddRule(rules ...Rule) {
+	s.rules = append(s.rules, rules...)
+}
+
+func (s *Scope) Rules() []Rule {
+	return s.rules
+}
+
+func (s *Scope) Disable() {
+	s.disabled = true
+}
+
+func (s *Scope) Enable() {
+	s.disabled = false
+}
+
+func (s *Scope) HandleHTTP(h http.Handler) func(http.ResponseWriter, *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if s.disabled {
+			h.ServeHTTP(w, r)
+			return
+		}
+
+		for _, rule := range s.rules {
+			if !rule.Match(r) {
+				// Continue
+				h.ServeHTTP(w, r)
+				return
+			}
+		}
+
+		s.plugins.Run(w, r, h)
+	}
+}
+
+type Manager struct {
+	Server   *http.Server
+	instance *vinxi.Vinxi
+	scopes   []*Scope
+}
+
+func Manage(instance *vinxi.Vinxi) *Manager {
+	m := &Manager{instance: instance}
+	instance.Layer.UsePriority("request", layer.Tail, m)
+	return m
+}
+
+func (m *Manager) NewScope(rules ...Rule) *Scope {
+	scope := NewScope(rules...)
+	m.scopes = append(m.scopes, scope)
+	return scope
+}
+
+func (a *Manager) HandleHTTP(w http.ResponseWriter, r *http.Request, h http.Handler) {
```

Comment:
Will fix. Code style is not too much relevant now, but thanks for highlighting this.